lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -1323,12 +1323,19 @@ def test_cube_setter(self):
         cube.fill_value = fill_value
         self.assertEqual(cube.fill_value, fill_value)
 
-    def test_cube_data_setter_with_none(self):
+    def test_cube_data_setter_with_ndarray(self):
         fill_value = 123
         cube = Cube(np.array(0), fill_value=fill_value)
         self.assertEqual(cube.fill_value, fill_value)
         cube.data = np.array(1)
-        self.assertEqual(cube.fill_value, None)
+        self.assertIsNone(cube.fill_value)
+
+    def test_cube_data_setter_with_masked_array(self):
+        fill_value = 123
+        cube = Cube(np.array(0))
+        self.assertIsNone(cube.fill_value)
+        cube.data = ma.masked_array(1, fill_value=fill_value)
+        self.assertEqual(cube.fill_value, fill_value)
```

Comment:
This doesn't check the data type of the fill value is correct but presumably the testing of that will be in the cube.fill_value setter